app/src/main/java/com/cp/tasky/auth/register/presentation/RegisterViewModel.kt

```diff
@@ -0,0 +1,100 @@
+package com.cp.tasky.auth.register.presentation
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.cp.tasky.auth.shared.domain.AuthRepository
+import com.cp.tasky.auth.shared.domain.model.UserRegisterBody
+import com.cp.tasky.auth.shared.domain.usecase.UserInputAuthValidationUseCase
+import com.cp.tasky.core.data.util.Error
+import com.cp.tasky.core.data.util.Result
+import com.cp.tasky.core.data.util.onError
+import com.cp.tasky.core.data.util.onSuccess
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class RegisterViewModel @Inject constructor(
+    private val authRepository: AuthRepository,
+) : ViewModel() {
+
+    private val userInputAuthValidationUseCase = UserInputAuthValidationUseCase()
+
+    var registerScreenState by mutableStateOf(RegisterViewState())
+        private set
+
+    val registerScreenNetworkState = MutableStateFlow<Result<Unit, Error>>(Result.Idle)
```

Comment:
Typically, we want to keep MutableStateFlows private in the ViewModel, since we want to prevent that external sources modify them (reduces bugs, since you only need to look for issues in one class). If you still need this flow in the UI, you can work with two properties:
```
private val _registerScreenNetworkState = MutableStateFlow<Result<Unit, Error>>(Result.Idle)
val registerScreenNetworkState = _registerScreenNetworkState.asStateFlow() // Read-only
```
And I think you could just call this `networkState` because it's obvious that this is for the register screen here, since the VM is bound to it :)